packages/ic-management/README.md

```diff
@@ -232,7 +233,23 @@ Creates a canister. Only available on development instances.
 | ------------------------------------- | ------------------------------------------------------------------------------------------------------- |
 | `provisionalCreateCanisterWithCycles` | `({ settings, amount, canisterId, }?: ProvisionalCreateCanisterWithCyclesParams) => Promise<Principal>` |
 
-[:link: Source](https://github.com/dfinity/ic-js/tree/main/packages/ic-management/src/ic-management.canister.ts#L296)
+[:link: Source](https://github.com/dfinity/ic-js/tree/main/packages/ic-management/src/ic-management.canister.ts#L300)
+
+##### :gear: ecdsaPublicKey
+
+Calculate a SEC1 encoded ECDSA public key for the given canister using the given derivation path. If the `canister_id` is unspecified, it will default to the canister id of the caller. The `derivation_path` is a vector of variable length byte strings. Each byte string may be of arbitrary length, including empty. The total number of byte strings in the `derivation_path` must be at most 255. The `key_id` is a struct specifying both a `curve` and a `name`. The availability of a particular `key_id` depends on implementation.
```

Comment:
> If the `canister_id` is unspecified, it will default to the canister id of the caller.

Wait, are you sure this endpoint can be called from a frontend client?
Or is this description incorrect?

I don't see how the canister can be identified if the calls happen on the cllient side.

Reply:
The documentation states the following:

> If the canister_id is unspecified, it will default to the canister id of the caller.

So, assuming it's just misleading, I would review the documentation.

Then, regarding your answer.

> Correct, it can be called without canister ID parameter and it will use the principal of the initialized service

Do you mean that it will use the principal of your snippet `canisterId: Principal.fromHex("")`? If so, I doubt that's correct or, again it's a misunderstanding and the documentation should state that if no parameter canisterId is provided, the address will be deferred from the canisterId on which the service lies - i.e. the Bitcoin canisterId.

Reply:
Agree, it is misleading. Check the new description